ide/xml.catalog/src/org/netbeans/modules/xml/catalog/settings/CatalogSettings.java

```diff
@@ -182,20 +182,20 @@ public final boolean isRemovable(CatalogReader provider) {
      * @return providers of given class or all if passed <code>null/code> argument.
      *         It never returns null.
      */
-    public final synchronized Iterator getCatalogs(Class[] providerClasses) {
+    public final synchronized Iterator<Object> getCatalogs(Class[] providerClasses) {
```

Comment:
Actually this is a `Iterator<? extends CatalogReader>` or `Iterator<CatalogReader>`. This quickly spins out of control - the big question: In your quest for warning less build - how far do you wanna go?